artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationSyncFileMessage.java

```diff
@@ -150,30 +161,54 @@ public void encodeRest(final ActiveMQBuffer buffer) {
          default:
             // no-op
       }
-
       buffer.writeInt(dataSize);
-      /*
-       * sending -1 will close the file in case of a journal, but not in case of a largeMessage
-       * (which might receive appends)
-       */
-      if (dataSize > 0) {
-         buffer.writeBytes(byteBuffer, 0, byteBuffer.writerIndex());
-      }
 
       release();
    }
 
    @Override
-   public void release() {
-      if (byteBuffer != null) {
-         byteBuffer.release();
-         byteBuffer = null;
+   public ActiveMQBuffer encode(CoreRemotingConnection connection) {
+      if (connection != null && connection.getTransportConnection() instanceof NettyConnection) {
+         ActiveMQBuffer buffer;
+         int bufferSize = expectedEncodeSize();
+         if (fileId != -1 && dataSize > 0) {
+            bufferSize -= dataSize;
+         }
+         buffer = createPacket(connection, bufferSize);
+
+         encodeHeader(buffer);
+
+         encodeRest(buffer, connection);
+
+         encodeSize(buffer, expectedEncodeSize());
+
+         return fileId != -1 && dataSize > 0 ? new FileDelegatedChannelBufferWrapper(file, offset, dataSize, buffer) : buffer;
```

Comment:
It implies that the file is being read lazily by Netty to apply the zero-copy transfert, but it means also that we need to be sure that the file won't be moved or deleted util the zero-copy transfert will finish

Reply:
The file is opend before sending and closed after finished. Transfer won't be affected if the file is removed or deleted bcs it's referenced?